tmail-backend/blob/blobid-list/src/main/java/com/linagora/tmail/blob/blobid/list/SingleSaveBlobStoreDAO.java

```diff
@@ -40,11 +41,16 @@ public InputStream read(BucketName bucketName, BlobId blobId) throws ObjectStore
         return blobStoreDAO.readBytes(bucketName, blobId);
     }
 
+    private boolean isBucketNameDefault(BucketName bucketName) {
+        Preconditions.checkNotNull(bucketName);
+        return defaultBucketName.equals(bucketName);
+    }
+
     @Override
     public Mono<Void> save(BucketName bucketName, BlobId blobId, byte[] data) {
         return Mono.from(blobIdList.isStored(blobId))
                 .flatMap(isStored -> {
-                    if (isStored) {
+                    if (isStored || !isBucketNameDefault(bucketName)) {
                         return Mono.empty();
```

Comment:
We should still save the blob in the blobstore when using non default buckets?

Reply:
https://github.com/linagora/tmail-backend/pull/203#discussion_r679775727

where you say: "If I remember correctly single save should be on only on the default bucket and do nothing for non-default buckets."

And now you telling me it should save the blob when not using default? Me confused :(

Or do you mean we should always save the blob to default bucket even when using an other one?

Reply:
Ok after discussion got it. single save -> only for default bucket. other buckets should keep a normal default blobstore behavior

Reply:
I don't understand. It is "||" (not "&&"), I think it will not change the old operation. Where Is my wrong?

Reply:
I misunderstood @chibenwa comment and most of this work will go to garbage, don't worry @vttranlina